sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj

```diff
@@ -21,10 +21,11 @@
 
   <ItemGroup>
     <Compile Include="..\..\..\eventgrid\Microsoft.Azure.WebJobs.Extensions.EventGrid\src\TriggerBinding\HttpRequestProcessor.cs" Link="EventGrid\HttpRequestProcessor.cs" />
-    <Compile Include="..\..\..\eventgrid\Microsoft.Azure.WebJobs.Extensions.EventGrid\src\TriggerBinding\SubscriptionValidationResponse.cs" Link="EventGrid\SubscriptionValidationResponse.cs" />
   </ItemGroup>
  
   <ItemGroup>
+    <!--Switch to package reference once next version is released-->
+    <ProjectReference Include="..\..\..\eventgrid\Azure.Messaging.EventGrid\src\Azure.Messaging.EventGrid.csproj" />
```

Comment:
I'm not sure we should be adding this as dependency.
We were talking the other day that it would be great to remove queues dependency and move that logic into host and expose it via some interface to extensions that need this to load balance. adding this new dependency would be a step back.
The event grid integration doesn't need whole event grid package.

@pakrym what do you think? you've been involved in these discussions.

Reply:
This dependency is needed to use the SubscriptionValidationResponse model. Do you have an alternative?

Reply:
I agree, bringing in the entire EG SDK just for this smallish feature seems heavy-handed. Can we mode the SubscriptionValidationResponse shared code into the storage SDK for now?

Reply:
The point of using the model from the SDK was in part to validate the scenario that customers would need to do for performing the handshake (by using an internal custom type, we didn't realize that we were missing a public ctor for the SDK model). We can ifdef this for storage but it seems correct to use the SDK model for the Event Grid extension.

Reply:
Storage user doesn't interact with EG much. They just set the trigger source like [here](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs#event-grid-strategy) - everything else happens behind the scenes.

Reply:
If this needs to be used by multiple services, we should look at moving it into Azure.Core rather than taking dependencies on EventGrid everywhere.  A shared source approach also works in the meantime.

Also, taking a dependency on EventGrid would [require explicit Arch Board approval.](https://azure.github.io/azure-sdk/general_implementation.html#general-dependencies-approved-only)

Reply:
We've already shipped this type as part of the Event Grid library. It is an Event Grid model - but the storage extensions use Event Grid under the hood.
I wanted to use the official Event Grid type in the extension to help ensure that the extension is doing the same thing a Webhook customer would do when using Event Grid.

Reply:
Is Storage the only case where we'd ever want to use this?  We'd have to look at the type carefully, but we might be able to shift it into Azure.Core without breaking (i.e., introduce a base class of the EG type in Azure.Core and move all the members up to it).

Reply:
Most likely it would only be needed for Storage, but it still feels like it would be weird to put in Core since it is specific to Event Grid (and the storage extension which uses Event Grid). For now, I've shadowed the type in Storage and removed the dependency.

Reply:
I don't think it's too weird to consider EG stuff in Azure.Core since we've already got the CloudEvent schema types and so many services use EG for surfacing notifications.  I'm happy to wait until we've got a third use for this though if we don't think it'll be common.